lib/analyse.ml

```diff
@@ -182,6 +185,35 @@ module Analysis = struct
       and+ ys = lwt_result_list_mapm ~f xs in
       y::ys
 
+  let opam_dep_file packages =
+    let lines =
+      [ {|opam-version: "2.0"|}; {|depends: [|} ]
+      @ List.map (fun (pkg, ver) -> Printf.sprintf {|  "%s" %s|} pkg ver) packages
+      @ [ {|]|} ]
+    in
+    lines |> List.map (fun s -> s ^ "\n") |> String.concat ""
+
+  let exactly v = Printf.sprintf {|{ = "%s" }|} v
+
+  let solver_cache = Hashtbl.create 128
+
+  let find_opam_repo_commit_for_ocamlformat ~solve ~platforms version =
+    let ( let+ ) = Lwt_result.Infix.( >|= ) in
+    let deps_for_ocamlformat =
+      ( "deps_for_ocamlformat.opam"
+      , opam_dep_file [("ocamlformat", exactly version)])
+    in
+    let+ workers =
+        match Hashtbl.find_opt solver_cache  platforms with
```

Comment:
Shouldn't this be keyed on `version`?

I don't think `platforms` needs to be in the key at all, although it could be stored with the result. Then if the result is for a different set of platforms you can reset the cache, because it's out of date.

Reply:
My mistake! However, should we really clear the cache if `platforms` mismatch? As some packages disable some of them in their opam file, wouldn't it have an impact or am I confusing two different things?

Reply:
`platforms` here is the input to the analysis, which should be the complete set. The solver may filter some of them out later.